src/AngleSharp.Css/Values/Composites/CssBackgroundSizeValue.cs

```diff
@@ -94,8 +109,14 @@ public String CssText
         /// </summary>
         /// <param name="other">The other background size.</param>
         /// <returns>True if both are equivalent, otherwise false.</returns>
-        public Boolean Equals(CssBackgroundSizeValue other) =>
-            _mode == other._mode && _height == other._height && _width == other._width;
+        public Boolean Equals(CssBackgroundSizeValue other)
+        {
+            if (_mode == ValueMode.Text && other._mode == ValueMode.Text)
```

Comment:
Unfortunately the _height/_width checks fail as it determines they are different objects. So had to make this a full function to bypass these checks for ValueMode.Text